colab-api/src/main/java/ch/colabproject/colab/api/controller/user/UserManager.java

```diff
@@ -201,78 +200,80 @@ public AuthMethod getAuthenticationMethod(String identifier) {
+     * {@link #signup(SignUpInfo) signup} method is called.
      *
-     * @param username      username
-     * @param email         email address
+     * @param signUpInfo    base of user data (username, firstname, lastname, email)
      * @param plainPassword plain text password
      *
-     * @return a brand new user
+     * @return a brand-new user
      *
      * @throws HttpErrorMessage if username is already taken
      */
-    public User createUser(String username, String email, String plainPassword) {
+    public User createUser(SignUpInfo signUpInfo, String plainPassword) {
         AuthMethod method = getDefaultRandomAuthenticationMethod();
-        SignUpInfo signUpinfo = new SignUpInfo();
 
-        signUpinfo.setUsername(username);
-        signUpinfo.setEmail(email);
-        signUpinfo.setHashMethod(method.getMandatoryMethod());
+        signUpInfo.setHashMethod(method.getMandatoryMethod());
 
-        signUpinfo.setSalt(method.getSalt());
+        signUpInfo.setSalt(method.getSalt());
 
         byte[] hash = method.getMandatoryMethod().hash(plainPassword, method.getSalt());
 
-        signUpinfo.setHash(Helper.bytesToHex(hash));
+        signUpInfo.setHash(Helper.bytesToHex(hash));
 
-        return this.signup(signUpinfo);
+        return this.signup(signUpInfo);
     }
 
     /**
-     * {@link #createAdminUser(java.lang.String, java.lang.String, java.lang.String)
-     * createAdminUser} within a brand new transaction.
+     * {@link #createAdminUser(String, String, String)
+     * createAdminUser} within a brand-new transaction.
      *
      * @param username      username
      * @param email         email address
      * @param plainPassword plain text password
      *
-     * @return a brand new user
+     * @return a brand-new user
      */
     @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
     public User createAdminUserTx(String username, String email, String plainPassword) {
         return this.createAdminUser(username, email, plainPassword);
     }
 
     /**
-     * Create a brand new admin user, which can authenticate with a {@link LocalAccount}. First,
+     * Create a brand-new admin user, which can authenticate with a {@link LocalAccount}. First,
      * plainPassword will be hashed as any client should do. Then the
-     * {@link #signup(ch.colabproject.colab.api.model.user.SignUpInfo) signup} method is called.
+     * {@link #signup(SignUpInfo) signup} method is called.
      *
      * @param username      username
      * @param email         email address
      * @param plainPassword plain text password
      *
-     * @return a brand new user
+     * @return a brand-new user
      *
      * @throws HttpErrorMessage if username is already taken
      */
     public User createAdminUser(String username, String email, String plainPassword) {
-        User admin = this.createUser(username, email, plainPassword);
+        SignUpInfo signUpInfo = new SignUpInfo();
```

Comment:
Why not pass signUpInfo directly like in createUser() ?

Reply:
Finally, I changed the structure of the method to be smarter